R/install_deps.R

```diff
@@ -1,29 +1,24 @@
-
 #' Install dependencies of reports within
 #'
-#' This function can be used to install package dependencies
-#' based on the reports within the factory.
-#'
-#' @export
-#'
-#' @author Thibaut Jombart \email{thibautjombart@@gmail.com}
-#'
+#' This function can be used to install package dependencies based on the
+#' reports within the factory.
+#' 
+#' @inheritParams compile_reports
 #' @param update A logical indicating if packages which are already installed
-#'   should be re-installed (\code{TRUE}); otherwise, only missing packages are
-#'   installed; defaults to \code{FALSE}.
-#'
-#' @param ... Arguments to be passed to \code{install.packages}.
-#'
+#'   should be re-installed (`TRUE`); otherwise, only missing packages are
+#'   installed; defaults to `FALSE`.
+#' @param ... Arguments to be passed to `install.packages()`.
+#' 
 #' @seealso \code{\link{list_deps}} to list dependencies of packages
-#'
-
-install_deps <- function(update = FALSE, ...) {
+#' 
+#' @export
+install_deps <- function(factory = ".", update = FALSE, ...) {
 
-  pkg_to_install <- list_deps(!update)
+  pkg_to_install <- list_deps(factory = factory, missing = !update)
 
   if (length(pkg_to_install) > 0L) {
```

Comment:
I tend to use `if (length(xxx))` nowadays, but cosmetic; just writing this to know if there's an emerging consensus on what's best.

Reply:
closed with https://github.com/reconhub/reportfactory/pull/68/commits/e60198f05faaf899a33aab96f3d9634ab3413252